scripts/calculateRevenue/protocols/utils/networks.ts

```diff
@@ -0,0 +1,59 @@
+import { QueryResponse, TransactionField } from '@envio-dev/hypersync-client'
+import { getHyperSyncClient } from '../../../utils'
+import { NetworkId } from '../../../types'
+
+export async function fetchTotalGasUsed({
+  networkId,
+  users,
+  startBlock,
+  endBlock,
+}: {
+  networkId: NetworkId
+  users: string[]
+  startBlock?: number
+  endBlock?: number
+}): Promise<number> {
+  let fromBlock = startBlock ?? 0
+  let totalGasUsed = 0
+  let hasMoreBlocks = true
+
+  try {
+    const client = getHyperSyncClient(networkId)
+
+    const query = {
```

Comment:
Is this query fetching all TXs a user has made on the network across a given block range?